### PR TITLE
[fix] update assignedFlavorIdx at the beginning of loop

### LINKED ISSUE #2391: Workloads are not admitted when there are 3 or more Workloads in ClusterQueue
**What happened**:

Sometimes, when there are 3+ Workloads waiting in ClusterQueue and they would fit in the quota, they are not admitted. They don't have any status or events. Reservation in quota doesn't happen. 

I increased `zap-log-level` to 6 and I can see this in logs, repeated over and over every ~2 seconds, even though it says 15 seconds in logs.:
image

It essentially schedules the workload, but it can take from X minutes to N hours...

Here's my configuration. I increased clientConnection.qps and burst to big values, but that didn't change anything. I also tried giving the deployment `kueue-controller-manager` unlimited resources, but it's barely using anything: ~20m CPU and ~100Mi memory.
```  controller_manager_config.yaml: |-
    apiVersion: config.kueue.x-k8s.io/v1beta1
    kind: Configuration
    health:
      healthProbeBindAddress: :8081
    metrics:
      bindAddress: :8080
    # enableClusterQueueResources: true
    webhook:
      port: 9443
    leaderElection:
      leaderElect: true
      resourceName: c1f6bfd2.kueue.x-k8s.io
    controller:
      groupKindConcurrency:
        Job.batch: 5
        Pod: 5
        Workload.kueue.x-k8s.io: 5
        LocalQueue.kueue.x-k8s.io: 1
        ClusterQueue.kueue.x-k8s.io: 1
        ResourceFlavor.kueue.x-k8s.io: 1
    clientConnection:
      qps: 5000
      burst: 20000
    #pprofBindAddress: :8082
    #waitForPodsReady:
    #  enable: true
    #manageJobsWithoutQueueName: true
    #internalCertManagement:
    #  enable: false
    #  webhookServiceName: ""
    #  webhookSecretName: ""
    integrations:
      frameworks:
      - "batch/job"
      - "pod"
      podOptions:
        namespaceSelector:
          matchExpressions:
            - key: kubernetes.io/metadata.name
              operator: NotIn
              values: [ "cattle-system", "calico-system", "cattle-monitoring-system", "cattle-logging-system", "cattle-provisioning-capi-system", "cattle-fleet-local-system", "cattle-fleet-system", "cattle-resources-system", "cattle", "calico", "nvidia", "tigera-operator", "label-studio", "postgres-ha", "minio-operator", "minio-tenant", "kube-system", "dlf", "directpv", "cert-manager", "longhorn-system", "postgresql-ha", "labelstudio", "coder", "harbor", "kueue-system", "kueue", "kyverno", "argocd" ]
        podSelector:
          matchExpressions:
          - key: app.kubernetes.io/name
            operator: In
            values: [ "coder-workspace" ]
```

Values (I cut the configMap as it's posted here above):
```enablePrometheus: true
controllerManager:
  replicas: 3
  featureGates:
    - name: QueueVisibility
      enabled: true
    - name: VisibilityOnDemand
      enabled: true
```

ClusterQueue:
```apiVersion: kueue.x-k8s.io/v1beta1
kind: ClusterQueue
metadata:
  name: cq-all-resources
spec:
  preemption:
    reclaimWithinCohort: Never
    withinClusterQueue: Never
  flavorFungibility:
    whenCanBorrow: Borrow
    whenCanPreempt: TryNextFlavor
  queueingStrategy: BestEffortFIFO
  namespaceSelector: {}
  resourceGroups:
    - coveredResources: ["cpu", "memory", "ephemeral-storage", "github.com/fuse", "pods", "nvidia.com/gpu"]
      flavors:
      - name: "hawking"
        resources:
        - name: "cpu"
          nominalQuota: 125
        - name: "memory"
          nominalQuota: 980Gi
        - name: "ephemeral-storage"
          nominalQuota: 9.85Ti
        - name: "github.com/fuse"
          nominalQuota: 5k
        - name: "pods"
          nominalQuota: 110
        - name: "nvidia.com/gpu"
          nominalQuota: 8
      - name: "euler"
        resources:
        - name: "cpu"
          nominalQuota: 61
        - name: "memory"
          nominalQuota: 488Gi
        - name: "ephemeral-storage"
          nominalQuota: 14Ti
        - name: "github.com/fuse"
          nominalQuota: 5k
        - name: "pods"
          nominalQuota: 110
        - name: "nvidia.com/gpu"
          nominalQuota: 9
      - name: "newton"
        resources:
        - name: "cpu"
          nominalQuota: 125
        - name: "memory"
          nominalQuota: 991Gi
        - name: "ephemeral-storage"
          nominalQuota: 19.7Ti
        - name: "github.com/fuse"
          nominalQuota: 5k
        - name: "pods"
          nominalQuota: 110
        - name: "nvidia.com/gpu"
          nominalQuota: 4
      - name: "godel"
        resources:
        - name: "cpu"
          nominalQuota: 253
        - name: "memory"
          nominalQuota: 980Gi
        - name: "ephemeral-storage"
          nominalQuota: 13.7Ti
        - name: "github.com/fuse"
          nominalQuota: 5k
        - name: "pods"
          nominalQuota: 110
        - name: "nvidia.com/gpu"
          nominalQuota: 8
      - name: "bayes"
        resources:
        - name: "cpu"
          nominalQuota: 125
        - name: "memory"
          nominalQuota: 991Gi
        - name: "ephemeral-storage"
          nominalQuota: 26636Gi
        - name: "github.com/fuse"
          nominalQuota: 5k
        - name: "pods"
          nominalQuota: 110
        - name: "nvidia.com/gpu"
          nominalQuota: 8
```

LocalQueue:
```apiVersion: kueue.x-k8s.io/v1beta1
kind: LocalQueue
metadata:
   namespace: rbenes
   name: lq-all-resources
spec:
   clusterQueue: cq-all-resources
```

Job:
```apiVersion: batch/v1
kind: Job
metadata:
  annotations:
    batch.kubernetes.io/job-tracking: ""
  creationTimestamp: "2024-06-10T14:05:14Z"
  generation: 1
  labels:
    app: ml-job
    kueue.x-k8s.io/priority-class: ml-medium
    kueue.x-k8s.io/queue-name: lq-all-resources
  name: ml-job-online-filtering-6-4f6f9f6
  namespace: rbenes
  resourceVersion: "771281792"
  uid: 2e40d4dd-48f3-48fc-9d42-577e6cbf9f12
spec:
  backoffLimit: 0
  completionMode: NonIndexed
  completions: 1
  parallelism: 1
  selector:
    matchLabels:
      batch.kubernetes.io/controller-uid: 2e40d4dd-48f3-48fc-9d42-577e6cbf9f12
  suspend: true
  template:
    metadata:
      creationTimestamp: null
      labels:
        batch.kubernetes.io/controller-uid: 2e40d4dd-48f3-48fc-9d42-577e6cbf9f12
        batch.kubernetes.io/job-name: ml-job-online-filtering-6-4f6f9f6
        controller-uid: 2e40d4dd-48f3-48fc-9d42-577e6cbf9f12
        job-name: ml-job-online-filtering-6-4f6f9f6
      name: ml-job-online-filtering-6-4f6f9f6
    spec:
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: kubernetes.io/hostname
                operator: NotIn
                values:
                - k8s-prod-master0
                - k8s-prod-master1
                - k8s-prod-master2
              - key: nvidia.com/gpu.memory
                operator: Gt
                values:
                - "4096"
      containers:
      - command:
        - bash
        - -c
        - |
          set -x &&
          cp /data/user/{.git*,clearml.conf} ~/ &&
          mkdir -p ~/.ssh &&
          cp /data/user/id_* ~/.ssh &&
          cp /data/user/.git* ~/ &&
          ssh-keyscan -p 7999 -t rsa foo.bar > ~/.ssh/known_hos
          cp ~/.ssh/id_rsa.pub ~/.ssh/authorized_keys &&
          chmod 600 ~/.ssh/id_rsa &&
          ls -lha ~ &&
          ls -lha ~/.ssh &&
          ls -la &&
          cd iris-recognition &&
          HYDRA_FULL_ERROR=1 python train.py &&
          export tag="$(python -m solver.config.get_config_value --key tag)" &&
          export task_dir="$(python -m solver.config.get_config_value --key directories.task_dir)" &&
          export last_ckpt_path=$task_dir/trainings/$tag/ckpt/last.ckpt &&
          HYDRA_FULL_ERROR=1 python post_evaluate.py --model ${last_ckpt_path} --gpu 0 1
        env:
        - name: WORKLOAD_ARGS
        - name: AWS_ACCESS_KEY_ID
          valueFrom:
            secretKeyRef:
              key: AWS_ACCESS_KEY_ID
              name: user-credentials
        - name: AWS_SECRET_ACCESS_KEY
          valueFrom:
            secretKeyRef:
              key: AWS_SECRET_ACCESS_KEY
              name: user-credentials
        - name: CLEARML_API_ACCESS_KEY
          valueFrom:
            secretKeyRef:
              key: CLEARML_API_ACCESS_KEY
              name: user-credentials
        - name: CLEARML_API_SECRET_KEY
          valueFrom:
            secretKeyRef:
              key: CLEARML_API_SECRET_KEY
              name: user-credentials
        image: foo:bar
        imagePullPolicy: Always
        name: ml-main
        resources:
          limits:
            cpu: "30"
            ephemeral-storage: 5Gi
            memory: 70Gi
            nvidia.com/gpu: "4"
          requests:
            cpu: "30"
            ephemeral-storage: 3Gi
            memory: 70Gi
            nvidia.com/gpu: "4"
        securityContext:
          privileged: true
          runAsGroup: 6000
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /data/user/
          name: user-configuration
        - mountPath: /srv/
          name: srv
        - mountPath: /mnt/srv.remote/
          name: srv-remote
        - mountPath: /mnt/sandbox/
          name: shared-storage
        - mountPath: /data/workspace/
          name: workspace
        - mountPath: /mnt/nas.brno/
          name: nas-brno
        - mountPath: /dev/shm
          name: dev-shm
        workingDir: /data/workspace
      dnsPolicy: ClusterFirst
      imagePullSecrets:
      - name: registry-credentials
      initContainers:
      - command:
        - bash
        - -c
        - |
          mkdir ~/.ssh &&
          ssh-keyscan -p 7999 -t rsa foo > ~/.ssh/known_hosts &&
          cp /data/user/id_* ~/.ssh/ &&
          chmod 600 ~/.ssh/id_rsa &&
          git lfs install &&
          git clone $GIT_REPO &&
          cd "$(basename "$GIT_REPO" .git)"
          if ! [ -z "$GIT_CHECKOUT_REF" ]; then
            git checkout "$GIT_CHECKOUT_REF"
          elif ! [ -z "$GIT_CHECKOUT_BRANCH" ]; then
            git checkout "$GIT_CHECKOUT_BRANCH"
          else
            echo "Please set either GIT_CHECKOUT_REF or GIT_CHECKOUT_BRANCH!"
            exit 1
          fi
          git submodule update --init
        env:
        - name: GIT_REPO
          value: foo:bar
        - name: GIT_CHECKOUT_REF
          value: 4f6f9f6
        - name: GIT_CLONE_PROTECTION_ACTIVE
          value: "false"
        image: bitnami/git:latest
        imagePullPolicy: Always
        name: init-git-repo
        resources:
          limits:
            cpu: 250m
            memory: 512Mi
          requests:
            cpu: 100m
            memory: 512Mi
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /data/user
          name: user-configuration
        - mountPath: /data/workspace
          name: workspace
        workingDir: /data/workspace
      priorityClassName: ml-medium
      restartPolicy: Never
      runtimeClassName: nvidia
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 30
      volumes:
      - emptyDir:
          medium: Memory
          sizeLimit: 2Gi
        name: dev-shm
      - name: user-configuration
        projected:
          defaultMode: 420
          sources:
          - secret:
              items:
              - key: id_rsa
                path: id_rsa
              - key: id_rsa.pub
                path: id_rsa.pub
              - key: .git-credentials
                path: .git-credentials
              name: user-credentials
          - configMap:
              items:
              - key: .gitconfig
                path: .gitconfig
              name: gitconfig
          - configMap:
              items:
              - key: clearml.conf
                path: clearml.conf
              name: clearml.conf
      - hostPath:
          path: /srv/
          type: ""
        name: srv
      - hostPath:
          path: /mnt/sandbox/
          type: ""
        name: shared-storage
      - hostPath:
          path: /mnt/srv.remote/
          type: ""
        name: srv-remote
      - emptyDir: {}
        name: workspace
      - hostPath:
          path: /mnt/nas.brno
          type: ""
        name: nas-brno
      - configMap:
          defaultMode: 420
          name: iengine.lic
        name: iengine-lic
  ttlSecondsAfterFinished: 1209600
status:
  conditions:
  - lastProbeTime: "2024-06-10T14:05:14Z"
    lastTransitionTime: "2024-06-10T14:05:14Z"
    message: Job suspended
    reason: JobSuspended
    status: "True"
    type: Suspended
  ready: 0
  uncountedTerminatedPods: {}
```

Workload:
```apiVersion: kueue.x-k8s.io/v1beta1
kind: Workload
metadata:
  creationTimestamp: "2024-06-10T14:05:14Z"
  finalizers:
  - kueue.x-k8s.io/resource-in-use
  generation: 1
  labels:
    kueue.x-k8s.io/job-uid: 2e40d4dd-48f3-48fc-9d42-577e6cbf9f12
  name: job-ml-job-online-filtering-6-4f6f9f6-9a073
  namespace: rbenes
  ownerReferences:
  - apiVersion: batch/v1
    blockOwnerDeletion: true
    controller: true
    kind: Job
    name: ml-job-online-filtering-6-4f6f9f6
    uid: 2e40d4dd-48f3-48fc-9d42-577e6cbf9f12
  resourceVersion: "771281796"
  uid: c7d282a7-8deb-493d-85dc-5ff4fee05b7b
spec:
  active: true
  podSets:
  - count: 1
    name: main
    template:
      metadata:
        name: ml-job-online-filtering-6-4f6f9f6
      spec:
        affinity:
          nodeAffinity:
            requiredDuringSchedulingIgnoredDuringExecution:
              nodeSelectorTerms:
              - matchExpressions:
                - key: kubernetes.io/hostname
                  operator: NotIn
                  values:
                  - k8s-prod-master0
                  - k8s-prod-master1
                  - k8s-prod-master2
                - key: nvidia.com/gpu.memory
                  operator: Gt
                  values:
                  - "4096"
        containers:
        - command:
          - bash
          - -c
          - |
            set -x &&
            cp /data/user/{.git*,clearml.conf} ~/ &&
            mkdir -p ~/.ssh &&
            cp /data/user/id_* ~/.ssh &&
            cp /data/user/.git* ~/ &&
            ssh-keyscan -p 7999 -t rsa foo > ~/.ssh/known_hos
            cp ~/.ssh/id_rsa.pub ~/.ssh/authorized_keys &&
            chmod 600 ~/.ssh/id_rsa &&
            ls -lha ~ &&
            ls -lha ~/.ssh &&
            ls -la &&
            cd iris-recognition &&
            HYDRA_FULL_ERROR=1 python train.py &&
            export tag="$(python -m solver.config.get_config_value --key tag)" &&
            export task_dir="$(python -m solver.config.get_config_value --key directories.task_dir)" &&
            export last_ckpt_path=$task_dir/trainings/$tag/ckpt/last.ckpt &&
            HYDRA_FULL_ERROR=1 python post_evaluate.py --model ${last_ckpt_path} --gpu 0 1
          env:
          - name: WORKLOAD_ARGS
          - name: AWS_ACCESS_KEY_ID
            valueFrom:
              secretKeyRef:
                key: AWS_ACCESS_KEY_ID
                name: user-credentials
          - name: AWS_SECRET_ACCESS_KEY
            valueFrom:
              secretKeyRef:
                key: AWS_SECRET_ACCESS_KEY
                name: user-credentials
          - name: CLEARML_API_ACCESS_KEY
            valueFrom:
              secretKeyRef:
                key: CLEARML_API_ACCESS_KEY
                name: user-credentials
          - name: CLEARML_API_SECRET_KEY
            valueFrom:
              secretKeyRef:
                key: CLEARML_API_SECRET_KEY
                name: user-credentials
          image: foo:bar
          imagePullPolicy: Always
          name: ml-main
          resources:
            limits:
              cpu: "30"
              ephemeral-storage: 5Gi
              memory: 70Gi
              nvidia.com/gpu: "4"
            requests:
              cpu: "30"
              ephemeral-storage: 3Gi
              memory: 70Gi
              nvidia.com/gpu: "4"
          securityContext:
            privileged: true
            runAsGroup: 6000
          terminationMessagePath: /dev/termination-log
          terminationMessagePolicy: File
          volumeMounts:
          - mountPath: /data/user/
            name: user-configuration
          - mountPath: /srv/
            name: srv
          - mountPath: /mnt/srv.remote/
            name: srv-remote
          - mountPath: /mnt/sandbox/
            name: shared-storage
          - mountPath: /data/workspace/
            name: workspace
          - mountPath: /mnt/nas.brno/
            name: nas-brno
          - mountPath: /dev/shm
            name: dev-shm
          workingDir: /data/workspace
        dnsPolicy: ClusterFirst
        imagePullSecrets:
        - name: registry-credentials
        initContainers:
        - command:
          - bash
          - -c
          - |
            mkdir ~/.ssh &&
            ssh-keyscan -p 7999 -t rsa foo > ~/.ssh/known_hosts &&
            cp /data/user/id_* ~/.ssh/ &&
            chmod 600 ~/.ssh/id_rsa &&
            git lfs install &&
            git clone $GIT_REPO &&
            cd "$(basename "$GIT_REPO" .git)"
            if ! [ -z "$GIT_CHECKOUT_REF" ]; then
              git checkout "$GIT_CHECKOUT_REF"
            elif ! [ -z "$GIT_CHECKOUT_BRANCH" ]; then
              git checkout "$GIT_CHECKOUT_BRANCH"
            else
              echo "Please set either GIT_CHECKOUT_REF or GIT_CHECKOUT_BRANCH!"
              exit 1
            fi
            git submodule update --init
          env:
          - name: GIT_REPO
            value: foo:bar
          - name: GIT_CHECKOUT_REF
            value: 4f6f9f6
          - name: GIT_CLONE_PROTECTION_ACTIVE
            value: "false"
          image: bitnami/git:latest
          imagePullPolicy: Always
          name: init-git-repo
          resources:
            limits:
              cpu: 250m
              memory: 512Mi
            requests:
              cpu: 100m
              memory: 512Mi
          terminationMessagePath: /dev/termination-log
          terminationMessagePolicy: File
          volumeMounts:
          - mountPath: /data/user
            name: user-configuration
          - mountPath: /data/workspace
            name: workspace
          workingDir: /data/workspace
        priorityClassName: ml-medium
        restartPolicy: Never
        runtimeClassName: nvidia
        schedulerName: default-scheduler
        securityContext: {}
        terminationGracePeriodSeconds: 30
        volumes:
        - emptyDir:
            medium: Memory
            sizeLimit: 2Gi
          name: dev-shm
        - name: user-configuration
          projected:
            defaultMode: 420
            sources:
            - secret:
                items:
                - key: id_rsa
                  path: id_rsa
                - key: id_rsa.pub
                  path: id_rsa.pub
                - key: .git-credentials
                  path: .git-credentials
                name: user-credentials
            - configMap:
                items:
                - key: .gitconfig
                  path: .gitconfig
                name: gitconfig
            - configMap:
                items:
                - key: clearml.conf
                  path: clearml.conf
                name: clearml.conf
        - hostPath:
            path: /srv/
            type: ""
          name: srv
        - hostPath:
            path: /mnt/sandbox/
            type: ""
          name: shared-storage
        - hostPath:
            path: /mnt/srv.remote/
            type: ""
          name: srv-remote
        - emptyDir: {}
          name: workspace
        - hostPath:
            path: /mnt/nas.brno
            type: ""
          name: nas-brno
        - configMap:
            defaultMode: 420
            name: iengine.lic
          name: iengine-lic
  priority: 100
  priorityClassName: ml-medium
  priorityClassSource: kueue.x-k8s.io/workloadpriorityclass
  queueName: lq-all-resources
```

This is happening for both plain Pods and Jobs. If I delete the top two waiting Workloads, the rest are admitted.

BTW this is how the ClusterQueue looks like (custom Grafana visualization):
image

**What you expected to happen**:
Workloads that fit into the quota gets admitted immediately.

**How to reproduce it (as minimally and precisely as possible)**:
Unfortunately, I wasn't able to reproduce it on out test cluster with the same configuration. It's not deterministic, as it doesn't happen everytime there are 3+ workloads in queue.

**Anything else we need to know?**:
It was also happening on versions v0.6.0/1/2/3.

**Environment**:
- Kubernetes version (use `kubectl version`): v1.27.10+rke2r1
- Kueue version (use `git describe --tags --dirty --always`): v0.7.0
- Cloud provider or hardware configuration: onprem VMs
- OS (e.g: `cat /etc/os-release`): Ubuntu 22.04.4 LTS
- Kernel (e.g. `uname -a`): 5.15.0-97-generic
- Install tools: ArgoCD
- Others:

### PR DESCRIPTION
**Pull request by KunWuLuan**
#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes #2391

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Review by trasc — COMMENTED** [review]
overall LGTM

File: pkg/scheduler/flavorassigner/flavorassigner.go [hunk 1]
```diff
@@ -404,6 +404,7 @@ func (a *FlavorAssigner) findFlavorForPodSetResource(
 	assignedFlavorIdx := -1
 	idx := a.wl.LastAssignment.NextFlavorToTryForPodSetResource(psID, resName)
 	for ; idx < len(resourceGroup.Flavors); idx++ {
+		assignedFlavorIdx = idx
```
Comment (mimowo) [inline pkg/scheduler/flavorassigner/flavorassigner.go:407]: Can you please cover this change with a unit or integration test?

**Review by mimowo — COMMENTED** [review]
lgtm, but would be great to have a test for it

File: pkg/scheduler/flavorassigner/flavorassigner.go (hunk 1 above)
Comment (alculquicondor) [reply pkg/scheduler/flavorassigner/flavorassigner.go:407]: also we should probably rename the variable to `attemptedFlavorIdx`, but that can be done in a follow up.

File: pkg/scheduler/flavorassigner/flavorassigner_test.go [hunk 2]
```diff
@@ -2409,7 +2409,7 @@ func TestAssignFlavors(t *testing.T) {
 				t.Errorf("e.assignFlavors(_).RepresentativeMode()=%s, want %s", repMode, tc.wantRepMode)
 			}
 
-			if diff := cmp.Diff(tc.wantAssignment, assignment, cmpopts.IgnoreUnexported(Assignment{}, FlavorAssignment{}), cmpopts.IgnoreFields(Assignment{}, "LastState"), cmpopts.IgnoreFields(FlavorAssignment{}, "TriedFlavorIdx")); diff != "" {
+			if diff := cmp.Diff(tc.wantAssignment, assignment, cmpopts.IgnoreUnexported(Assignment{}, FlavorAssignment{}), cmpopts.IgnoreFields(Assignment{}, "LastState")); diff != "" {
```
Comment (mimowo) [inline pkg/scheduler/flavorassigner/flavorassigner_test.go:2412]: +1, would any of the pre-existing test cases fail after the change? If not please add a dedicated one.

Comment (KunWuLuan) [reply pkg/scheduler/flavorassigner/flavorassigner_test.go:2412]: yes, "each podset requires preemption on a different flavor" will fail after the change.
```
Unexpected assignment (-want,+got):
          flavorassigner.Assignment{
          	PodSets: []flavorassigner.PodSetAssignment{
          		{
          			Name: "launcher",
          			Flavors: flavorassigner.ResourceAssignment{
          				s"cpu": &{
          					Name:           "one",
          					Mode:           s"Preempt",
        - 					TriedFlavorIdx: -1,
        + 					TriedFlavorIdx: 0,
          					... // 1 ignored field
          				},
          			},
          			Status:   &{reasons: {...}},
          			Requests: {s"cpu": {i: {...}, Format: "DecimalSI"}},
          			Count:    1,
          		},
          		{Name: "workers", Flavors: {s"cpu": &{Name: "tainted", Mode: s"Preempt", TriedFlavorIdx: -1}}, Status: &{reasons: {...}}, Requests: {s"cpu": {i: {...}, Format: "DecimalSI"}}, ...},
          	},
          	Borrowing: false,
          	... // 2 ignored and 1 identical fields
          }
```

For launcher, flavor 1 is tainted, so in the past triedFlavor idx would be 0 and now triedFlavor idx would be -1.

Comment (mimowo) [reply pkg/scheduler/flavorassigner/flavorassigner_test.go:2412]: ack

**Comment by mimowo** [comment]
/lgtm
/approve
Thanks!

**Comment by mimowo** [comment]
/release-note-edit
```release-note
Fix for the scenario when a workload which is not matching a certain resource flavor
is not checked against the consecutive resource flavors.
```

**Comment by alculquicondor** [comment]
/release-note-edit

```release-note
Fix for the scenario when a workload doesn't match some resource flavors due to affinity or taints could cause the workload to be continuously retried.
```

**Pull request merged**